src/python/gudhi/persistence_graphical_tools.py

```diff
@@ -107,7 +107,7 @@ def plot_persistence_barcode(
     alpha=0.6,
     max_intervals=20000,
     inf_delta=0.1,
-    legend=False,
+    legend=True,
```

Comment:
For the diagram, the legend appears at the bottom right, where there are no points anyway, so it is always good. For the badcode, it is more likely to hide relevant bars. Maybe with a location of 'lower left' or 'best' it would be less likely to overlap.

Reply:
Set to 'best' for barcode and 'lower right' for diagrams